context/http.go

```diff
@@ -103,8 +103,13 @@ func GetRequestID(ctx Context) string {
 // WithResponseWriter returns a new context and response writer that makes
 // interesting response statistics available within the context.
 func WithResponseWriter(ctx Context, w http.ResponseWriter) (Context, http.ResponseWriter) {
+	closeNotifier, ok := w.(http.CloseNotifier)
+	if !ok {
+		panic("the ResponseWriter does not implement CloseNotifier")
```

Comment:
If the overridden `CloseNotify` method finds that `instrumentedResponseWriter.closeNotifier` is nil, should it panic then?

It feels a little wrong to satisfy the `CloseNotifier` interface, but possibly panic when the exposed method is called. Though it doesn't make a practical difference within the context of docker/distribution, since the type assertion should always succeed.